v2/yarpcrouter/router.go

```diff
@@ -72,6 +74,56 @@ func NewMapRouter(defaultService string, rs []yarpc.TransportProcedure) MapRoute
 	return router
 }
 
+// EncodingToTransportProcedures converts encoding-level procedures to transport-level procedures.
+func EncodingToTransportProcedures(encodingProcedures []yarpc.EncodingProcedure) ([]yarpc.TransportProcedure, error) {
+	transportProcedures := make([]yarpc.TransportProcedure, len(encodingProcedures))
+	for i, p := range encodingProcedures {
+		var transportHandlerSpec yarpc.TransportHandlerSpec
+		switch p.HandlerSpec.Type() {
+		case yarpc.Unary:
+			transportHandlerSpec = yarpc.NewUnaryTransportHandlerSpec(&unaryTransportHandler{p})
+		default:
+			return nil, fmt.Errorf("unknown handler spec type: %v", p.HandlerSpec.Type())
+		}
+
+		transportProcedures[i] = yarpc.TransportProcedure{
+			Name:        p.Name,
+			Service:     p.Service,
+			HandlerSpec: transportHandlerSpec,
+			Encoding:    p.Encoding,
+			Signature:   p.Signature,
+		}
+	}
+
+	return transportProcedures, nil
+}
+
+// Allows encoding-level procedures to act as transport-level procedures.
+type unaryTransportHandler struct {
+	h yarpc.EncodingProcedure
+}
+
+func (u *unaryTransportHandler) Handle(ctx context.Context, req *yarpc.Request, reqBuf *yarpc.Buffer) (*yarpc.Response, *yarpc.Buffer, error) {
+	res := &yarpc.Response{}
+	decodedBody, err := u.h.Codec.Decode(reqBuf)
+	if err != nil {
+		return res, nil, err
+	}
+
+	body, err := u.h.HandlerSpec.Unary().Handle(ctx, decodedBody)
+	if err != nil {
+		res.ApplicationError = true
+		return res, nil, err
+	}
+
+	encodedBody, err := u.h.Codec.Encode(body)
+	if err != nil {
+		return res, nil, err
+	}
+
+	return res, encodedBody, nil
+}
```

Comment:
This is really cool 👍

Reply:
I'm not sure this counts as "coming from the InboundCodec". Also, what about the headers? 
@AlexAPB

Reply:
Looks like I was following behind you in #1615! Feel free to ignore my last comment :) This is awesome stuff